### PR TITLE
Improve coverage sycl interface

### PR DESCRIPTION
**Pull request by oleksandr-pavlyk**
This PR extends `test_sycl_queue_submit` to use 1D range, 2D range, 3D range and use dependent events.

- [x] Have you provided a meaningful PR description?
- [x] Have you added a test, reproducer or referred to an issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you opening the PR as a draft?

### CODE REVIEW

**Comment by oleksandr-pavlyk** [comment]
Coverage info: https://coveralls.io/builds/66187084 

image

**Comment by oleksandr-pavlyk** [comment]
Additional changes pushed:

- Submit for float64 should only be exercised if device has needed aspect (otherwise coverage and tests can not be run on Iris Xe)
- submit local_accessor is now done for instances with ndim=1, 2, 3 getting coverage of `dpctl_sycl_interface.cpp` to over 90%
- Added test for `DPCTLQueue_Create` when context ref is nullptr, but device is not to cover few more lines


Changes to `tests/CMakeLists.txt`:
- Created target `run-c-api-tests` to run tests with environment variable set
- Created target `llvm-cov-report` to output per-file coverage summary and `llvm-cov-show` to output per-line coverage info (handly to detect which lines are not covered)
- Removed `llvm-cov` (it is now effectively `llvm-cov-report`)
- Each `llvm-cov-report`, `llvm-cov-show` and `llvm-genhtml` depend on `run-c-api-tests`

Changes to `dbg_build.sh`:
- Added lines to set `PATH` and `LLVM_TOOLS_HOME` env. variable so that cmake can find `llvm-cov` and other programs
- Replaced `make llvm-genhtml with `make llvm-cov-report` in the script.

File: libsyclinterface/tests/test_sycl_device_aspects.cpp [hunk 1]
```diff
@@ -97,7 +97,7 @@ auto build_gtest_values(const std::array<std::pair<T1, T2>, N> &params)
 auto build_params()
 {
     constexpr auto param_1 = get_param_list<const char *>(
-        "opencl:gpu", "opencl:cpu", "level_zero:gpu", "host");
```
Comment (diptorupd) [inline libsyclinterface/tests/test_sycl_device_aspects.cpp:100]: Not for this PR, but the design where a list of filter string is used is an anti-pattern. I got rid of a similar pattern in numba-dpex unit tests last year. All tests should run on default device and we can control the default device using the ONEAPI_DEVICE_SELECTOR environment variable.

File: libsyclinterface/tests/test_sycl_queue_submit.cpp [hunk 2]
```diff
@@ -334,9 +356,11 @@ TEST_F(TestQueueSubmit, CheckForFloat)
 
 TEST_F(TestQueueSubmitFP64, CheckForDouble)
 {
-    submit_kernel<double>(QRef, KBRef, spirvBuffer_, spirvFileSize_,
-                          DPCTLKernelArgType::DPCTL_FLOAT64_T,
-                          "_ZTS11RangeKernelIdE");
+    if (DPCTLDevice_HasAspect(DRef, DPCTLSyclAspectType::fp64)) {
```
Comment (diptorupd) [inline libsyclinterface/tests/test_sycl_queue_submit.cpp:359]: Good catch

**Review by diptorupd — APPROVED** [review]

**Pull request merged**